sync/src/chain.rs

```diff
@@ -950,7 +950,10 @@ impl ChainSync {
 			nonce: chain.nonce(a),
 			balance: chain.balance(a),
 		};
-		let _ = self.miner.import_transactions(transactions, fetch_account);
+		let res = self.miner.import_transactions(transactions, fetch_account);
+		if res.is_ok() {
```

Comment:
It updates the block only if all transactions were imported successfuly. It should rather be `if any transaction was imported successfuly`.